### PR TITLE
[SPARK-44451][BUILD] Make built document downloadable

### PR DESCRIPTION
**Pull request by zhengruifeng**
### What changes were proposed in this pull request?
Make built document downloadable

new steps:
image

built-docs:
image

(actually 55M to download)

image

### Why are the changes needed?
1, to make it much easier for reviewers to verify the document PRs;
~~2, to make it much easier for maintainers to check the built document in branches;~~

### Does this PR introduce _any_ user-facing change?
no

### How was this patch tested?
CI and manually check the downloaded docs

### CODE REVIEW

**Comment by zhengruifeng** [comment]
https://github.com/zhengruifeng/spark/actions/runs/5566016597/jobs/10167115410

I also test directly upload the directory, but it is slower and the artifact size is bigger (~112M):

image

image

(actually 112M to download)

**Comment by zhengruifeng** [comment]
cc @HyukjinKwon @dongjoon-hyun @Yikun @LuciferYang

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -686,6 +686,14 @@ jobs:
         fi
         cd docs
         bundle exec jekyll build
+        cd ..
```
[the remark this answers is not in the document]
Comment (zhengruifeng) [reply .github/workflows/build_and_test.yml:689]: ok, let me have a try

[the remark this answers is not in the document]
Comment (HyukjinKwon) [reply .github/workflows/build_and_test.yml:689]: Yeah, I think we don't need this.

File: .github/workflows/build_and_test.yml [hunk 2]
```diff
@@ -686,6 +686,14 @@ jobs:
...
+    - name: Tar documentation
+      run: tar cjf site.tar.bz2 docs/_site
+    - name: Upload documentation
+      uses: actions/upload-artifact@v3
```
[the remark this answers is not in the document]
Comment (zhengruifeng) [reply .github/workflows/build_and_test.yml:693]: that depends on whether we need to monitor the build-doc in apache/repo.
I personally want to also upload in apache/repo, since the cost seems quietly small.

File: .github/workflows/build_and_test.yml [hunk 3]
```diff
@@ -686,6 +686,14 @@ jobs:
...
+      with:
+        name: site
```
[the remark this answers is not in the document]
Comment (zhengruifeng) [reply .github/workflows/build_and_test.yml:695]: good point, I am not aware about the quota limited.

[the remark this answers is not in the document]
Comment (zhengruifeng) [reply .github/workflows/build_and_test.yml:694]: added this config, thanks for pointing it out

File: dev/merge_spark_pr.py [hunk 4]
```diff
@@ -44,7 +44,7 @@
 
 # Location of your Spark git development area
 SPARK_HOME = os.environ.get("SPARK_HOME", os.getcwd())
-# Remote name which points to the Gihub site
+# Remote name which points to the Github site
```
Comment (LuciferYang) [inline dev/merge_spark_pr.py:47]: This is correct, but should it be in an independent pr?

Comment (zhengruifeng) [reply dev/merge_spark_pr.py:47]: @LuciferYang ok, let me send a separate minor PR for it.

Comment (zhengruifeng) [reply dev/merge_spark_pr.py]: https://github.com/apache/spark/pull/42032

File: .github/workflows/build_and_test.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (zhengruifeng) [reply .github/workflows/build_and_test.yml:692]: on second thought, I think we can add this condition first, to be more conservative

**Review by HyukjinKwon — APPROVED** [review]
I think this is fine.

**Review by LuciferYang — APPROVED** [review]

**Review by Yikun — APPROVED** [review]
Thanks, LGTM.

**Comment by zhengruifeng** [comment]
thank you all

merged to master (4.0.0)
